pkger/service.go

```diff
@@ -1796,29 +1794,31 @@ func (s *Service) applyNotificationEndpoints(ctx context.Context, userID influxd
 		}
 
 		mutex.Do(func() {
-			endpoints[i].id = influxEndpoint.GetID()
-			for _, secret := range influxEndpoint.SecretFields() {
-				switch {
-				case strings.HasSuffix(secret.Key, "-routing-key"):
-					if endpoints[i].parserEndpoint.routingKey == nil {
-						endpoints[i].parserEndpoint.routingKey = new(references)
-					}
-					endpoints[i].parserEndpoint.routingKey.Secret = secret.Key
-				case strings.HasSuffix(secret.Key, "-token"):
-					if endpoints[i].parserEndpoint.token == nil {
-						endpoints[i].parserEndpoint.token = new(references)
-					}
-					endpoints[i].parserEndpoint.token.Secret = secret.Key
-				case strings.HasSuffix(secret.Key, "-username"):
-					if endpoints[i].parserEndpoint.username == nil {
-						endpoints[i].parserEndpoint.username = new(references)
+			if influxEndpoint != nil {
+				endpoints[i].id = influxEndpoint.GetID()
+				for _, secret := range influxEndpoint.SecretFields() {
+					switch {
+					case strings.HasSuffix(secret.Key, "-routing-key"):
+						if endpoints[i].parserEndpoint.routingKey == nil {
+							endpoints[i].parserEndpoint.routingKey = new(references)
+						}
+						endpoints[i].parserEndpoint.routingKey.Secret = secret.Key
+					case strings.HasSuffix(secret.Key, "-token"):
+						if endpoints[i].parserEndpoint.token == nil {
+							endpoints[i].parserEndpoint.token = new(references)
+						}
+						endpoints[i].parserEndpoint.token.Secret = secret.Key
+					case strings.HasSuffix(secret.Key, "-username"):
+						if endpoints[i].parserEndpoint.username == nil {
+							endpoints[i].parserEndpoint.username = new(references)
+						}
+						endpoints[i].parserEndpoint.username.Secret = secret.Key
+					case strings.HasSuffix(secret.Key, "-password"):
+						if endpoints[i].parserEndpoint.password == nil {
+							endpoints[i].parserEndpoint.password = new(references)
+						}
+						endpoints[i].parserEndpoint.password.Secret = secret.Key
```

Comment:
not sure how go likes to roll, but it doesn't look like there's a `default` here. does this handle all possible inputs?